pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -189,36 +205,48 @@ func (t *Tracer[Event]) Attach(pid uint32, eventCallback func(*Event)) error {
 		return nil
 	}
 
-	a, err := newAttachment(pid, netns, t.socketEnricher, t.spec,
-		t.bpfProgName, t.bpfPerfMapName, t.bpfSocketAttach)
+	a, err := t.newAttachment(pid, netns)
 	if err != nil {
 		return fmt.Errorf("creating network tracer attachment for pid %d: %w", pid, err)
 	}
 	t.attachments[netns] = a
 
-	go t.listen(netns, a.perfRd, t.baseEvent, t.processEvent, eventCallback)
-
 	return nil
 }
 
 func (t *Tracer[Event]) SetEventHandler(handler any) {
+	if t.eventHandler != nil {
+		panic("handler already set")
+	}
+
 	nh, ok := handler.(func(ev *Event))
 	if !ok {
 		panic("event handler invalid")
 	}
 	t.eventHandler = nh
+	go t.listen(t.perfRd, t.baseEvent, t.processEvent, t.eventHandler)
+}
+
+func (t *Tracer[Event]) EventCallback(event any) {
+	e, ok := event.(*Event)
+	if !ok {
+		panic("event handler argument invalid")
+	}
+	if t.eventHandler == nil {
+		return
+	}
+	t.eventHandler(e)
 }
```

Comment:
I think it's kind of legacy to support `pkg/gadget-collection`, if that's right, can we put a comment stating that?